ironfish-rust-nodejs/src/lib.rs

```diff
@@ -164,6 +177,11 @@ impl ThreadPoolHandler {
 }
 
 #[napi]
-pub fn is_valid_public_address(hex_address: String) -> bool {
-    PublicAddress::from_hex(&hex_address).is_ok()
+pub fn is_valid_public_address(address: JsBuffer) -> Result<bool> {
+    let address_buffer = address.into_value()?;
+    let address_vec = address_buffer.as_ref();
+    let mut address_bytes = [0; PUBLIC_ADDRESS_SIZE];
+    address_bytes.clone_from_slice(&address_vec[0..PUBLIC_ADDRESS_SIZE]);
```

Comment:
I was getting some test failures here coming from the validator tests. It was panicking I'm guessing from here because there was an out of bounds error. Maybe a shorter one would not validate here? I think it might be panicking on the longer ones too I'm not sure

Reply:
Added a length check